docs/kubernetes-large-clusters.md

```diff
@@ -0,0 +1,51 @@
+# Large Kubernetes Clusters with acs-engine
+
+## Background
+Starting from acs-engine v0.3.0, acs-engine supports using exponential cloud backoff that is a feature of Kubernetes v1.6.6 and newer. Cloud backoff allows Kubernetes nodes to backoff on HTTP 429 errors that are usually caused by exceeding Azure API limits.
+
+## To Use
+Declare your kubernetes cluster API model config as you normally would, with the following requirements:
+- Make sure you're declaring version 1.6.6 of Kubernetes (the only currently supported version that includes the configuration vectors described in [examples/largeclusters/kubernetes.json](https://github.com/Azure/acs-engine/blob/master/examples/largeclusters/kubernetes.json)). Kubernetes v1.6.6 is the default Kubernetes version starting in acs-engine v0.3.0.
+- We recommend the use of smaller pools (e.g., count of 20) over larger pools (e.g., count of 100); produce your desired total node count with lots of pools, as opposed to as few as possible.
+- We also recommend using large vmSize configurations: there's no intrinsic benefit to more nodes (in fact you could argue generally that there is intrinsic value to having fewer nodes in terms of reduced cluster complexity); make sure there's a defensible reason for scaling up capacity by way of increasing k8s node count instead of just using more powerful k8s node infrastructure. As Kubernetes excels in binpacking pods onto available instances, vertically scaling VM sizes (more CPU/RAM) is better approach for expanding cluster capacity.
```

Comment:
Let's modify this statement a bit to include acknowledgement that there are conditions under which more nodes are simply required. E.g., see the 100 pod per node limit here:

https://kubernetes.io/docs/admin/cluster-large/

How about:

```- We also recommend using large vmSize configurations to reduce node counts, where appropriate. Make sure you have a defensible infrastructure justification for more nodes in terms of node count, instead of opting to use more powerful nodes (for example as of kubernetes 1.7 there is a 100 pods per node limit). Doing so reduces cluster complexity, and azure resource administrative overhead. As Kubernetes excels in binpacking pods onto available instances, vertically scaling VM sizes (more CPU/RAM) is a better approach for expanding cluster capacity, if you are not approaching the pod-per-node limit.```